### ISSUE TITLE
Knowledge distillation: A good teacher is patient and consistent

### ISSUE THREAD

**Issue by ghuijo** [One-Day-One-Paper/One-Day-One-Paper#229]
### 정리
* model을 compression하는 새로운 방법을 제시하는 건 아니고 기존 knowledge disillation 과정을 비교하고 가장 classificatinon accuracy가 높은 학습 전략에 대해 연구함
* ImageNet-21k 데이터셋으로 pretrain되고 기타 다른 데이터셋으로 finetuning된 큰 BiT-ResNet-152x2 모델을 compress해 작은 사이즈의 데이터셋 범위에서 정확도 손실 없이 standard ResNet-50 architecture로 distill함.
1. Make sure that teacher and student always get identical inputs, including noise
2. Introduce aggressive data augmentations to enrich the input image manifold (through mixup)
3. Use very long training schedules

### consistent teaching
* various design choices when doing knowledge distillation
  <image src="https://user-images.githubusercontent.com/67627471/178147928-fd6d2df3-aed8-454a-8a81-9abeb1ad07bc.png" width=80% height=80%>
* Fixed teacher: precomputed target으로 teacher는 고정된 이미지를 받아서 teacher의 예측이 주어진 이미지에 대해 constant하고, student는 랜덤하게 augement된 이미지를 받음.
* Independent noise: teacher와 student가 각자 독립된 augmented 이미지를 받음.
* Consistent teaching: student와 teacher 모델이 같은 augmented 이미지를 받음. mild random cropping과 heavy inception crop 중 하나의 방법으로 이미지를 한 번만 랜덤하게 크롭함.
* Function matching: consistent teaching을 확장한 방법으로 본 논문에서 FunMatch라고 부름. 같은 augmented 이미지를 받는데, mixup 방식을 사용해 augmentation됨.

### patient teaching
* 트레이닝 에포크 수를 크게 해 학습시키는 방법
* Training duration에 따른 Test accuracy 비교
  <image src="https://user-images.githubusercontent.com/67627471/178147967-a564ff5f-188c-45be-9fa3-29242f56158c.png" width=80% height=80%>
  * patient하고 consist하게 학습시킨 ResNet-50 student 모델이 가장 우수한 성능을 보여주었고, expensive ResNet-152x2 teacher을 능가함

### Result
* Top-1 accuracy on ImageNet
  <image src="https://user-images.githubusercontent.com/67627471/178147975-dd5e5059-0b22-4f7a-9db4-1d05d37e2c1f.png" width=30% height=30%>
